yapapi/services.py

```diff
@@ -191,26 +204,143 @@ def _respond_nowait(self, message: Optional[Any], response_to: Optional[ServiceS
     async def get_payload() -> Optional[Payload]:
         """Return the payload (runtime) definition for this service.
 
+        To be overridden by the author of a specific Service class.
+
         If `get_payload` is not implemented, the payload will need to be provided in the
         `Golem.run_service` call.
         """
         pass
 
-    async def start(self):
-        """Implement the `starting` state of the service."""
+    async def start(self) -> AsyncGenerator[WorkItem, Awaitable[List[events.CommandEvent]]]:
+        """Implement the handler for the `starting` state of the service.
+
+        To be overridden by the author of a specific Service class.
+
+        Should perform the minimum set of operations after which the instance of a service can be
+        treated as "started", or, in other words, ready to receive service requests. It's up to the
+        developer of the specific Service class to decide what exact operations constitute a
+        service startup.
+
+        As a handler, it's expected to be a generator that yields `WorkItems` (generated using the
```

Comment:
I would remove "As a handler" because this explains nothing to me, I think this is something meaningful only for those who know `yapapi` internals. I might be missing something.

Reply:
yes, that's exactly the meaning and I disagree with this suggestion ... I'll think how to adjust it though